### ISSUE TITLE
v2.4.7/8 bug

### ISSUE THREAD

**Issue by s7dhansh** [jagi/meteor-astronomy#633]
I am not sure what exactly is causing this. Will try to figure it out when I get some time.
In the meanwhile, if you can get something out of this error, it would be great!

The same code works fine with 2.4.6. `setSeriousness` is a meteorMethod in a collection and the same Class has an afterSave event, which seems to be triggering this error.

```
 => awaited here:
    at Promise.await (/opt/s.portal/app/programs/server/npm/node_modules/meteor/promise/node_modules/meteor-promise/promise_server.js:60:12)
    at [object Object]._.extend.apply (packages/ddp-server/livedata_server.js:1624:22)
    at callStandardMeteorMethod (packages/jagi:astronomy/lib/modules/storage/utils/callStandardMeteorMethod.js:7:21)
    at Class.applyMethod (packages/jagi:astronomy/lib/modules/methods/class_prototype_methods/applyMethod.js:18:10)
    at Class.setSeriousness (packages/jagi:astronomy/lib/modules/methods/utils/wrapMethod.js:23:16)
    at Class.create.events.afterSave (common/imports/api/group/collection.js:398:33)
    at packages/jagi:astronomy/lib/modules/events/class_prototype_methods/dispatch_event.js:43:5
    at arrayEvery (/opt/s.portal/app/programs/server/npm/node_modules/meteor/jagi_astronomy/node_modules/lodash/_arrayEvery.js:16:10)
    at every (/opt/s.portal/app/programs/server/npm/node_modules/meteor/jagi_astronomy/node_modules/lodash/every.js:53:10)
    at Class.dispatchEvent (packages/jagi:astronomy/lib/modules/events/class_prototype_methods/dispatch_event.js:42:3)
    at triggerAfterSave (packages/jagi:astronomy/lib/modules/storage/utils/trigger_after_save.js:5:6)
    at documentUpdate (packages/jagi:astronomy/lib/modules/storage/utils/document_update.js:97:5)
    at Class.save (packages/jagi:astronomy/lib/modules/storage/class_prototype_methods/save.js:144:20)
    at Class.create.meteorMethods.setSeriousness (common/imports/api/group/collection.js:493:10)
    at [object Object].astronomyExecute (packages/jagi:astronomy/lib/modules/methods/meteor_methods/astronomyExecute.js:34:17)

```

**Comment by lukejagodzinski** [comment]
It's hard to tell without reproduction. It looks like the problem is in the `meteor-promise` package.

**Comment by s7dhansh** [comment]
The problem arises due to a recursion happening when I run a meteorMethod in afterSave which saves the same document. It happened when I created a new document, after which afterSave ran, modified the doc and then went into an infinite loop.

I am not sure if it is a bug or a problem with my code. But somehow, this code had no issues earlier. Still, I secured my code by moving the change on `afterUpdate` instead of `afterSave`, and the problem has gone.

I will try to dig deep later, but for now I will close this.
